pkg/controller/gcp/database/claim.go

```diff
@@ -75,27 +91,44 @@ func ConfigureMyCloudsqlInstance(_ context.Context, cm resource.Claim, cs resour
 		return errors.Errorf("expected instance claim %s to be %s", cm.GetName(), databasev1alpha1.MySQLInstanceGroupVersionKind)
 	}
 
-	rs, csok := cs.(*corev1alpha1.ResourceClass)
+	rs, csok := cs.(*v1alpha1.CloudsqlInstanceClass)
 	if !csok {
-		return errors.Errorf("expected resource class %s to be %s", cs.GetName(), corev1alpha1.ResourceClassGroupVersionKind)
+		return errors.Errorf("expected resource class %s to be %s", cs.GetName(), v1alpha1.CloudsqlInstanceClassGroupVersionKind)
 	}
 
 	i, mgok := mg.(*v1alpha1.CloudsqlInstance)
 	if !mgok {
 		return errors.Errorf("expected managed resource %s to be %s", mg.GetName(), v1alpha1.CloudsqlInstanceGroupVersionKind)
 	}
 
-	spec := v1alpha1.NewCloudSQLInstanceSpec(rs.Parameters)
+	spec := &v1alpha1.CloudsqlInstanceSpec{
+		ResourceSpec: corev1alpha1.ResourceSpec{
+			ReclaimPolicy: corev1alpha1.ReclaimRetain,
+		},
+		CloudsqlInstanceParameters: rs.SpecTemplate.CloudsqlInstanceParameters,
+	}
+
 	translated := translateVersion(my.Spec.EngineVersion, v1alpha1.MysqlDBVersionPrefix)
 	v, err := resource.ResolveClassClaimValues(spec.DatabaseVersion, translated)
 	if err != nil {
 		return err
 	}
 	spec.DatabaseVersion = v
 
+	// NOTE(hasheddan): consider moving defaulting to either CRD or managed reconciler level
```

Comment:
is this the same defaulting as done for postgres CloudSQL? if so, want to make it a single reusable function so we are only doing this in one place?

Reply:
That sounds like a good pattern to me!